app/ts/components/subcomponents/AutosizingInput.tsx

```diff
@@ -55,15 +55,12 @@ function ParsedInput<T>(model: ParsedInputModel<T>) {
 	// we want to pass through all model values *except* the rawValue, which may contain a password
 	const inputModel = { ...model }
 	delete inputModel.rawValue
-	return <input { ...inputModel } class = 'autosizing-input' value = { internalValue } onInput = { event => internalValue.value = event.currentTarget.value } onChange = { onChange }/>
+	return <input { ...inputModel } class = 'autosizing-input' value = { internalValue } onInput = { event => { internalValue.value = event.currentTarget.value } } onChange = { onChange }/>
 }
 
 function Input<T>(model: UnparsedInputModel | ParsedInputModel<T>) {
-	if ('tryParse' in model && model.tryParse) {
-		return <ParsedInput { ...model }/>
-	} else {
-		return <ParsedInput { ...model } value = { new OptionalSignal(model.value)} sanitize = { model.sanitize || (x => x) } tryParse = { value => ({ ok: true, value }) } serialize = { x => x || '' }/>
-	}
+	if ('tryParse' in model && model.tryParse) return <ParsedInput { ...model }/>
+	return <ParsedInput { ...model } value = { new OptionalSignal(model.value)} sanitize = { model.sanitize || (x => x) } tryParse = { value => ({ ok: true, value }) } serialize = { x => x || '' }/>
```

Comment:
I like the if/else better, since neither code path is an "early exit", they are just two equally valid code paths.

I like one-line returns when it is a guard clause, not when it is a logical branch.